catalog/app/containers/NavBar/Search.js

```diff
@@ -34,72 +47,178 @@ const useStyles = M.makeStyles((t) => ({
+}) {
   const {
-    adornment,
     disabled: disabledCls,
-    iconized: iconizedCls,
+    expanded: expandedCls,
+    help: helpCls,
     hidden: hiddenCls,
+    iconized: iconizedCls,
+    inputIcon: inputIconCls,
+    inputOptions: inputOptionsCls,
+    wrapper: wrapperCls,
     ...classes
   } = useStyles()
   return (
-    <M.InputBase
-      startAdornment={
-        <M.InputAdornment className={adornment}>
-          <M.Icon>search</M.Icon>
-        </M.InputAdornment>
-      }
-      classes={classes}
-      className={cx({
-        [disabledCls]: disabled,
-        [iconizedCls]: iconized,
-        [hiddenCls]: hidden,
-      })}
-      placeholder={
-        focused ? `Search ${bucket ? `s3://${bucket}` : 'all buckets'}` : 'Search'
-      }
-      disabled={disabled}
-      {...props}
-    />
+    <M.ClickAwayListener onClickAway={onHelpClose}>
+      <div className={wrapperCls}>
+        {helpOpened && (
+          <M.MuiThemeProvider theme={style.appTheme}>
+            <SearchHelp className={helpCls} onQuery={onQuery} />
+          </M.MuiThemeProvider>
+        )}
+
+        <M.InputBase
+          startAdornment={
+            <M.InputAdornment>
+              {iconized && !expanded ? (
+                <M.Icon className={inputIconCls} onClick={onToggleOptions}>
+                  search
+                </M.Icon>
+              ) : (
+                <Lab.ToggleButton
+                  className={inputOptionsCls}
+                  size="small"
+                  value="help"
+                  selected={helpOpened}
+                  onChange={onToggleOptions}
+                >
+                  <M.Icon size="small">search</M.Icon>
+                  <M.Icon size="small">
+                    {helpOpened ? 'arrow_drop_up' : 'arrow_drop_down'}
+                  </M.Icon>
+                </Lab.ToggleButton>
+              )}
+            </M.InputAdornment>
+          }
+          classes={classes}
+          className={cx({
+            [expandedCls]: expanded,
+            [disabledCls]: disabled,
+            [iconizedCls]: iconized,
+            [hiddenCls]: hidden,
+          })}
+          placeholder={
+            expanded ? `Search ${bucket ? `s3://${bucket}` : 'all buckets'}` : 'Search'
+          }
+          disabled={disabled}
+          {...props}
+        />
+      </div>
+    </M.ClickAwayListener>
   )
 }
 
 function State({ query, makeUrl, children, onFocus, onBlur }) {
   const dispatch = redux.useDispatch()
 
   const [value, change] = React.useState(null)
-  const [focused, setFocused] = React.useState(false)
+  const [expanded, setExpanded] = React.useState(false)
+  const [helpOpened, setHelpOpened] = React.useState(false)
 
   const onChange = React.useCallback((evt) => {
     change(evt.target.value)
   }, [])
 
+  const handleExpand = React.useCallback(() => {
+    if (expanded) {
+      return
+    }
+
+    change(query)
+    setExpanded(true)
+    if (onFocus) onFocus()
+  }, [expanded, query])
```

Comment:
ok, setters may be skipped from the dependency list (as stated in the docs), but the callbacks must be stated there